### DISCUSSION TITLE
Checkbox value always 0 until cast defined

### DISCUSSION THREAD

**Discussion by dcoding1** [orchidsoftware/platform#2733] [Q&A]
Hello, i have strange checkbox behavior. 

What i have :
1) a checkbox :
`CheckBox::make('page.visible')
                    ->title('Visible')
                    ->help('if a page visible'),`

2) field defined in db as :
`$table->boolean('visible')->default(false);`

3) After saving a form, i get data :
`$data = $request->get('page');`

Case 1 : If checkbox was checked (db value=1) and i uncheck it and then send, works correct (no values passed).
Case 2 : If checkbox was unchecked (db value=0) and I check it and then send, works incorrect ( value=0 )
But, if I add casting to the model, Case 2 starts working correct, but value = 'on' instead of '1' :
`protected $casts = [
        'visible' => 'int'
    ];`
    
Case 3 : Checkboxes without model field always have value 'on'.

Anybody knows why so ?
Thanks.

**Comment by c-gross** [comment]
`int` is not defined in Laravel casting. 
Laravel Docs: _The supported cast types are: integer, real, float, double, string, boolean, object, array, collection, date, datetime, and timestamp._

I'm using `integer` in the model cast but boolean should also work.

**Reply by dcoding1** [reply]
yes, my bad, but it **integer** did not change anything, value is still 0. And i see that input always have attribute **value="0"**, no matter if I use ->value(1). So wondering why there is **value="0"** in html ?

**Comment by tabuna** [comment]
Hi @dcoding1, I think you need the `sendTrueOrFalse` method. 


```php
CheckBox::make('page.visible')
    ->sendTrueOrFalse()
    ->title('Visible')
    ->help('if a page visible'),
```


HTML forms are not submitted by default if they are unchecked: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox
![image](https://github.com/orchidsoftware/platform/assets/5102591/07f0bcb8-2f15-4a37-ad4c-b405c80476c5)

**Reply by dcoding1** [reply]
Thanks bro, with this method value now is 1 as expected. But even though, I think that there is something wrong. Because i had **value=0** for **checked** checkbox. And i believe, that it has to send **1** to server without setting any method as it is default checkbox functionality.
